python/ray/tune/execution/tune_controller.py

```diff
@@ -1539,6 +1539,13 @@ def _schedule_trial_stop(self, trial: Trial, exception: Optional[Exception] = No
 
         logger.debug(f"Requesting to STOP actor for trial {trial}")
 
+        if self._cached_trial_decisions.get(trial.trial_id):
+            logger.debug(
+                f"Trial {trial} is currently saving/pausing. Scheduling STOP after "
+                f"save resolved."
+            )
+            self._cached_trial_decisions[trial.trial_id] = TrialScheduler.STOP
```

Comment:
Should we instead check `if trial.is_saving`? it's ok for a trial that's PAUSING to get elevated to STOPPING

Reply:
I think it doesn't make a difference in practice as `_cached_trial_decisions` is only set when the trial is saving.

OTOH with `trial.is_saving` it means that if a scheduler manually schedules a save and then a stop, it will be a cached decision and executed after the stop resolved, which may be what we want. So yes, this might make more sense. Updated!